### DISCUSSION TITLE
Sofar HYD 10ktl-3ph support

### DISCUSSION THREAD

**Discussion by juraurban** [StephanJoubert/home_assistant_solarman#104] [Q&A]
I would like to integrate my new pv system with homeassistant, but apparently have different Sofar inverter model than is supported with this integration. Data logger LSW3 and serial number 2348486334.

Hope someone can help me with this.

I have found file with register address - tried to change some in config file but did not work.
[SofarHYD Modbus protocol explaination-3ph.xlsx](https://github.com/StephanJoubert/home_assistant_solarman/files/9623920/SofarHYD.Modbus.protocol.explaination-3ph.xlsx)

Thanks

**Comment by igieon** [comment]
@juraurban did you try this my file https://github.com/StephanJoubert/home_assistant_solarman/blob/main/custom_components/solarman/inverter_definitions/sofar_g3hyd.yaml . I look into xls and itseems similar what i have it and based that file try put it into cusomtom componets and see if you get data.

**Reply by psasin** [reply]
Thanks for the suggestion. 
I use also inverter **SofarSolar HYD 10KTL-3PH** and with **sofar_g3hyd.yaml** it is working just fine. 
Just correct lookup file must be selected when you are creating integration in HA. You can't change lookup file when integration is already created. In this case You must delete integration and create new integration with correct lookup file.
This is webpage of my Dataloger:
![2023-01-29 09_06_40-Mozilla Firefox](https://user-images.githubusercontent.com/40150768/215314627-d2279ad8-6cf2-48ce-a196-30dbd1458a73.png)
This is settings I have made in HA:
![2023-01-29 09_24_55-Nastavení – Home Assistant — Mozilla Firefox](https://user-images.githubusercontent.com/40150768/215314635-15c7222b-c206-4222-9a3c-a169ca1f6209.png)

**Reply by danielstuevel** [reply]
I also have this inverter. Unfortunately, I am not getting any values back. I have tested the connection via terminal. It is open. 
I have deleted the configuration and reconfigured it. unfortunately without success. 
It looks a little different on the Dataloger page. Wireless AP mode and STA mode are enabled. But without showing SSID.  Under the IP, however, the port is ‘open’.
Are there any other ways/tools to test the connection?

**Reply by psasin** [reply]
> I also have this inverter. Unfortunately, I am not getting any values back. I have tested the connection via terminal. It is open. I have deleted the configuration and reconfigured it. unfortunately without success. It looks a little different on the Dataloger page. Wireless AP mode and STA mode are enabled. But without showing SSID. Under the IP, however, the port is ‘open’. Are there any other ways/tools to test the connection?

You have to see Your SSID in Router SSID. Otherwise HomeAssistnat can not acccess the invertorr because it is not connected to the network. Configuration should be done thru Wizard page in Dataloger. Wizard will show You available SSIDs. You just select Your SSID.

![image](https://github.com/user-attachments/assets/85e01528-66a1-42e4-8e6c-320c28b3183b)

**Reply by danielstuevel** [reply]
I don't think that's the problem. I received an open response from the Home Assistant terminal with ‘nc -zv 192.168.1.58 8899’. So the connection seems to be established and the service is available.

**Reply by danielstuevel** [reply]
Have you been able to integrate your inverter in the meantime?

**Reply by MichaelPloese** [reply]
I am experiencing the same issue after my inverter HYD15ktl-3ph was upgraded to latest firmware rel 12. The integration can connect and sees the logger + entities, but the entities don´t return a value... Tried to delete and reinstall alreday, but the problem is keeping...

Could you solve your issue in the meantime?

**Reply by danielstuevel** [reply]
Unfortunately not. I am now following this post and waiting for news. https://github.com/StephanJoubert/home_assistant_solarman/issues/723
